src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl

```diff
@@ -65,6 +64,19 @@ function Models.default_initial_conditions(model::Nonhydrostatic2DModel)
     return Fields.FieldVector(; zero_inits...)
 end
 
+function Models.default_ode_cache(model::Nonhydrostatic2DModel, space_center, space_face)
+
+    local_geometry_center = Fields.local_geometry_field(space_center)
+
+    FT = eltype(model.domain)
+    zero_instance = zero(FT) # somehow need this, otherwise eltype inference error
+
+    q_liq = map(coord -> zero_instance, local_geometry_center)
+    q_ice = map(coord -> zero_instance, local_geometry_center)
```

Comment:
We could initialize cache names somewhere as (similar to initilize prog varnames) and loop over here as in the default init condition function.

Reply:
That sounds good. I'll see if I can do this.

Reply:
Would it be possible to assemble the cache using also BaseModelStyles and ThermoStyles? For example a base mode even without other equations added needs pressure in the cache.